lab_4_retrieval_w_clustering/main.py

```diff
@@ -314,8 +487,25 @@ def get_clusters_info(self, num_examples: int) -> list[dict[str, int | list[str]
             num_examples (int): Number of examples for each cluster
 
         Returns:
-            list[dict[str, int| list[str]]]: List with information about each cluster
-        """
+            list[dict[str, int | list[str]]]: List with information about each cluster
+        """
+        if not (isinstance(num_examples, int) and num_examples > 0):
+            raise ValueError
+        list_of_cluster_info = []
+        for cluster_index, cluster in enumerate(self.__clusters):
+            info_dict: dict[str, int | list[str]] = {"cluster_id": cluster_index}
```

Comment:
why do you add typing?

Reply:
I tried doing like this:
info_dict: {"cluster_id": cluster_index}
[modify info_dict]
return info_dict
But MyPy doesn't like it that non-integer values are added to the dict that previously only contained integer value(s)

Reply:
I tried to create a stub value for the "documents" key (see "Trying to fix MyPy 2"), but it didn't work. Typing seems like the best solution.